### PR TITLE
[ADD] data cleaning: overview

### PR DESCRIPTION
**Pull request by samueljlieber**
Task: [#3979113](https://www.odoo.com/odoo/project/3835/tasks/3979113?cids=3) 

This PR introduces the **Data Cleaning** application's documentation to the **Productivity** scope. This PR specifically details an overview of the app, only detailing the required modules and the Deduplication dashboard. More details for the other parts of the application will be handled in another PR.

### CODE REVIEW

**Comment by samueljlieber** [comment]
Hi @ksc-odoo! Requesting your peer review on this brand new Data Cleaning PR! Thank you in advance! 🙏

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:166]:
```suggestion
  - :guilabel:`Case/Accent Insensitive Match`: the characters in the text match, regardless of casing
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:170]:
```suggestion
   The rule does **not** capture duplicates without at least one deduplication rule set in the :guilabel:`Deduplication Rules` field.
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:172]:
```suggestion
With the rule's configuration complete, either close the rule form, or :ref:`run the rule manually
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:180]:
```suggestion
To manually run a specific deduplication rule at any time, navigate to :menuselection:`Data Cleaning app
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:181]:
```suggestion
--> Configuration --> Rules: Deduplication`, and select the rule to run.
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:191]:
```suggestion
On the :guilabel:`Field Recycle Records` dashboard (:menuselection:`Data Cleaning app --> Recycle Records`), Odoo
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:192]:
```suggestion
detects records that can be archived or deleted, by matching conditions within the records set by the
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:193]:
```suggestion
recycle record's rules.
```
please disregard if not accurate

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:198]:
```suggestion
On the :guilabel:`Field Cleaning Records* dashboard (:menuselection:`Data Cleaning app --> Field Cleaning`), Odoo
```

File: content/applications/productivity/data_cleaning.rst
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:207]:
```suggestion
The :guilabel:`Merge Action Manager` (:menuselection:`Data Cleaning app --> Configuration --> Merge Action
```
I wasn't able to locate this during my revisions. Might be worth double-checking to ensure the accuracy of this portion.

Screenshot 2024-06-21 090356

**Review by ksc-odoo — APPROVED** [review]
Hey @samueljlieber -- just finished my Peer Review on this PR. Great work! Super detailed. Just have a handful of comments/feedback for you to consider during your revisions. But, since they're all very manageable, I am approving now! 👍 Feel free to move this to the next stage, once you implement all the necessary adjustments. Thanks!

File: content/applications/productivity/data_cleaning.rst
Comment (samueljlieber) [reply content/applications/productivity/data_cleaning.rst:11]: Yeah I prefer using bold formatting here when introducing and defining these features only because they are in a list & colon. Having these features in italics makes it difficult to read when in a layout such as this IMO.

Thanks for you– definitely considered!

File: content/applications/productivity/data_cleaning.rst
Comment (samueljlieber) [reply content/applications/productivity/data_cleaning.rst:38]: Oops! Nice catch– yes the module is actually titled: "**Data Cleaning (merge)**"

File: content/applications/productivity/data_cleaning.rst
Comment (samueljlieber) [reply content/applications/productivity/data_cleaning.rst:137]: Yes– except the entered title is replaced once a **Model** is selected... so I decided to just explain selecting a Model as the first step, since this will add a default title that the user can _then_ replace/update :)

**Comment by samueljlieber** [comment]
Thank you so much for your insightful review @ksc-odoo! I implemented most of your suggestions in f06d814.

Passing this PR along to @StraubCreative for a final content review, TIA!

**Comment by samueljlieber** [comment]
Rebased to latest in 1fe6f31

File: content/applications/productivity/data_cleaning.rst
Comment (StraubCreative) [inline content/applications/productivity/data_cleaning.rst:11]: Consider turning the labels into internal anchors to respective sections on the page.

**Comment by samueljlieber** [comment]
Thank you so much for your helpful review @StraubCreative! I implemented your suggestions in 9fce69a.

I asked to update the code owner in the doc channel :) Will merge after the team is created.

**Comment by samueljlieber** [comment]
@robodoo r+

**Pull request closed without merging**
